hackathon_site/hardware/test_api.py

```diff
@@ -493,8 +500,28 @@ def test_user_has_no_view_permissions(self):
         response = self.client.get(self.view)
         self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
 
-    def test_user_has_view_permissions(self):
-        self._login(self.permissions)
+    def test_user_has_view_permissions_no_profile(self):
+        self._login(self.view_permissions)
+        response = self.client.get(self.view)
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_user_has_view_permissions_with_profile(self):
+        Profile.objects.create(user=self.user, team=self.team)
+        self._login(self.view_permissions)
+        response = self.client.get(self.view)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        queryset = Order.objects.all().filter(team_id=self.team.id)
+
+        # need to provide a request in the serializer context to produce absolute url for image field
+        expected_response = OrderListSerializer(
+            queryset, many=True, context={"request": response.wsgi_request}
+        ).data
+        data = response.json()
+
+        self.assertEqual(expected_response, data["results"])
+
+    def test_user_has_change_permissions(self):
```

Comment:
I left a comment on the view as well, but get requests should be governed only by view permissions